lib/datadog/appsec/contrib/active_record/patcher.rb

```diff
@@ -19,27 +19,43 @@ def target_version
           end
 
           def patch
-            ActiveSupport.on_load :active_record do
+            patch_proc = lambda do |adapter_class, is_postgres: false|
```

Comment:
I think it was the very first attempt to make it that way with postgres being an argument of a function. I think here is an issue which is band-aided. I think it's better to rethink and simplify patching instead of quick-fixing it, WDYT?